### PR TITLE
cmd: small refactor rule init code

### PR DESCRIPTION
**Pull request by i582**
The `Config.Rules` field is now fully initialized, including
the fields of the `rules.Set` structure. Thus, we got rid of the
need to initialize this field every time.

The `InitEmbeddedRules` function has been renamed to `AddEmbeddedRules`
and now accepts not the config, but the rules directly, as well
as for the `appendRuleSet` function.

### CODE REVIEW

File: src/cmd/rules.go [hunk 1]
```diff
@@ -8,19 +8,19 @@ import (
 	"strings"
 
 	"github.com/VKCOM/noverify/src/cmd/embeddedrules"
-	"github.com/VKCOM/noverify/src/linter"
 	"github.com/VKCOM/noverify/src/rules"
 )
 
-func InitEmbeddedRules(config *linter.Config, p *rules.Parser, filter func(r rules.Rule) bool) ([]*rules.Set, error) {
+func AddEmbeddedRules(rules *rules.Set, p *rules.Parser, filter func(r rules.Rule) bool) ([]*rules.Set, error) {
```
Comment (quasilyte) [inline src/cmd/rules.go]: You can call this param `rset`.

Comment (i582) [reply src/cmd/rules.go]: `rset` is already used inside the loop

File: src/linter/conf.go [hunk 2]
```diff
@@ -68,7 +68,7 @@ func NewConfig() *Config {
 	addBuiltinCheckers(reg)
 	return &Config{
 		SrcInput:       inputs.NewDefaultSourceInput(),
-		Rules:          &rules.Set{},
+		Rules:          rules.NewSet(),
```
Comment (quasilyte) [inline src/linter/conf.go:71]: There is a problem here.
cloneRulesForFile from linter/utils.go will now copy rules sets even if rules are not used.
We need to change `ruleSet != nil` check with something else as they're never nil now.

File: src/cmd/rules.go [hunk 3]
```diff
@@ -8,20 +8,21 @@ import (
...
-	ruleSets, err := parseEmbeddedRules(p)
+func AddEmbeddedRules(rset *rules.Set, p *rules.Parser, filter func(r rules.Rule) bool) ([]*rules.Set, error) {
+	embeddedRules, err := parseEmbeddedRules(p)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, rset := range ruleSets {
-		appendRuleSet(config, rset, filter)
+	for _, embeddedRule := range embeddedRules {
```
Comment (quasilyte) [inline src/cmd/rules.go:21]: It's not a rule, but a rule set.
Maybe remove the embeddedRule variable and use `embeddedRules[i]` to pass it as an argument?

Comment (i582) [reply src/cmd/rules.go]: Can be renamed to `embeddedRuleSet` and `embeddedRules` to `embeddedRuleSets`.

**Review by quasilyte — APPROVED** [review]

**Pull request merged**